Android app crashes when put in the background

- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Bug in mobile app**

**Describe the bug**

The Android application crashes when put in the background and then reopened. I don't even need to log into an account to make the application crash.

**To Reproduce**

Steps to reproduce the behavior:

1. Open the Tutanota Android application.
2. Go back to the launcher.
3. Reopen the Tutanota application.
4. The application crashes and is relaunched automatically by the OS, which means that I end up on the home screen, logged out.

**Expected behavior**

Not to crash :slightly_smiling_face: 

**Screenshots**

N/A

**Smartphone (please complete the following information):**

 - Device: Pixel 3a
 - OS: GrapheneOS (Android)
 - Version: Android 12

**Additional context**

The logs related to the crash:

```
06-08 21:07:11.794  4858  4858 D MainActivity: onStop                                                                                                                                         
06-08 21:07:11.844  4858  4858 E JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 719284)                                                                                        
06-08 21:07:11.846  4858  4858 W ActivityStopInfo: Bundle stats:                                                                                                                              
06-08 21:07:11.848  4858  4858 W ActivityStopInfo:   WEBVIEW_CHROMIUM_STATE [size=717132]                                                                                                     
06-08 21:07:11.848  4858  4858 W ActivityStopInfo: PersistableBundle stats:                                                                                                                   
06-08 21:07:11.848  4858  4858 W ActivityStopInfo:   [null]                                                                                                                                   
06-08 21:07:11.849  4858  4858 D AndroidRuntime: Shutting down VM                                                                                                                             
06-08 21:07:11.850  4858  4858 E AndroidRuntime: FATAL EXCEPTION: main                                                                                                                        
06-08 21:07:11.850  4858  4858 E AndroidRuntime: Process: de.tutao.tutanota, PID: 4858                                                                                                        
06-08 21:07:11.850  4858  4858 E AndroidRuntime: java.lang.RuntimeException: android.os.TransactionTooLargeException: data parcel size 719284 bytes                                           
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.app.ActivityClient.activityStopped(ActivityClient.java:87)                                                                 
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.app.servertransaction.PendingTransactionActions$StopInfo.run(PendingTransactionActions.java:143)                           
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:938)                                                                                
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)                                                                                
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)                                                                                        
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)                                                                                            
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7876)                                                                          
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)                                                                                     
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)                                                  
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at com.android.internal.os.ExecInit.main(ExecInit.java:49)                                                                            
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)                                                                
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)                                                                     
06-08 21:07:11.850  4858  4858 E AndroidRuntime: Caused by: android.os.TransactionTooLargeException: data parcel size 719284 bytes                                                            
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.os.BinderProxy.transactNative(Native Method)                                                                               
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.os.BinderProxy.transact(BinderProxy.java:571)
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.app.IActivityClientController$Stub$Proxy.activityStopped(IActivityClientController.java:1315)
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        at android.app.ActivityClient.activityStopped(ActivityClient.java:84)
06-08 21:07:11.850  4858  4858 E AndroidRuntime:        ... 11 more
06-08 21:07:11.864  1158  1878 W ActivityTaskManager:   Force finishing activity de.tutao.tutanota/.MainActivity
```

We can see that the application crashes when the `onStop` Android method is called.

--- a/app-android/app/src/main/java/de/tutao/tutanota/MainActivity.kt
+++ b/app-android/app/src/main/java/de/tutao/tutanota/MainActivity.kt
@@ -9,7 +9,6 @@ import android.graphics.Color
 import android.graphics.Rect
 import android.net.MailTo
 import android.net.Uri
-import android.os.Build
 import android.os.Bundle
 import android.os.PowerManager
 import android.provider.Settings
@@ -44,9 +43,6 @@ import kotlinx.coroutines.withContext
 import kotlinx.serialization.json.Json
 import kotlinx.serialization.json.jsonObject
 import kotlinx.serialization.json.jsonPrimitive
-import okhttp3.ConnectionSpec
-import okhttp3.OkHttpClient
-import org.conscrypt.Conscrypt
 import org.json.JSONException
 import org.json.JSONObject
 import java.io.File
@@ -54,13 +50,9 @@ import java.io.IOException
 import java.io.UnsupportedEncodingException
 import java.net.URLEncoder
 import java.security.SecureRandom
-import java.security.Security
 import java.util.*
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.TimeUnit
-import javax.net.ssl.SSLContext
-import javax.net.ssl.TrustManager
-import javax.net.ssl.X509TrustManager
 import kotlin.coroutines.Continuation
 import kotlin.coroutines.resume
 import kotlin.coroutines.resumeWithException
@@ -418,11 +410,6 @@ class MainActivity : FragmentActivity() {
 		}
 	}
 
-	override fun onSaveInstanceState(outState: Bundle) {
-		super.onSaveInstanceState(outState)
-		webView.saveState(outState)
-	}
-
 	suspend fun askBatteryOptimizationsIfNeeded() {
 		val powerManager = getSystemService(POWER_SERVICE) as PowerManager
 
@@ -621,7 +608,9 @@ class MainActivity : FragmentActivity() {
 			// Intent documentation claims that data is copied to ClipData if it's not there
 			// but we want to be sure
 			if (Intent.ACTION_SEND_MULTIPLE == intent.action) {
-				@Suppress("UNCHECKED_CAST")
+				// unchecked_cast: we could just check for null instead?
+				// deprecation: the alternative requires API 33
+				@Suppress("UNCHECKED_CAST", "DEPRECATION")
 				val uris = intent.extras!!.getParcelableArrayList<Uri>(Intent.EXTRA_STREAM)
 				if (uris != null) {
 					for (uri in uris) {
@@ -629,6 +618,8 @@ class MainActivity : FragmentActivity() {
 					}
 				}
 			} else if (intent.hasExtra(Intent.EXTRA_STREAM)) {
+				// depreciation: the alternative requires API 33
+				@Suppress("DEPRECATION")
 				val uri = intent.getParcelableExtra<Uri>(Intent.EXTRA_STREAM)
 				filesArray.add(uri.toString())
 			} else if (intent.data != null) {
@@ -665,6 +656,10 @@ class MainActivity : FragmentActivity() {
 		commonNativeFacade.openCalendar(userId)
 	}
 
+	// this still works, but there's onBackPressedDispatcher.addCallback
+	// it should work on all API levels we support:
+	// https://stackoverflow.com/questions/72634225/onbackpressed-is-deprecated-what-is-the-alternative
+	@Deprecated("Deprecated in Java")
 	override fun onBackPressed() {
 		if (commonSystemFacade.initialized) {
 			lifecycleScope.launchWhenCreated {
@@ -723,7 +718,6 @@ class MainActivity : FragmentActivity() {
 		const val OPEN_USER_MAILBOX_USERID_KEY = "userId"
 		const val IS_SUMMARY_EXTRA = "isSummary"
 		private const val ASKED_BATTERY_OPTIMIZATIONS_PREF = "askedBatteryOptimizations"
-		private const val MIGRATED_OLD_LOCAL_STORAGE_PREF = "migratedOldLocalStorage"
 		private const val TAG = "MainActivity"
 		private var requestId = 0
 

